extra/2-piping.js

```diff
@@ -16,26 +16,56 @@
   the final result to the variable goodCode
 */
 
-function add() {
+// function add() {
 
+// }
+
+function add(numberOne, numberTwo) {
+  return numberOne + numberTwo;
 }
 
-function multiply() {
+// const add = (numberOne, numberTwo) => numberOne + numberTwo;
+
+// function multiply() {
+
+// }
 
+function multiply(numberOne, numberTwo) {
+  return numberOne * numberTwo;
 }
 
-function format() {
+// const multiply = (numberOne, numberTwo) => numberOne * numberTwo;
+
+// function format() {
+
+// }
 
+function format(number) {
+  return `£${number}`
 }
 
+// const format = (number) => `£${number}`
+
 const startingValue = 2;
 
 // Why can this code be seen as bad practice? Comment your answer.
-let badCode =
+let badCode = format(multiply(add(startingValue, 10), 2));
+// this is calling 3 functions 2 of which are nested, its bad practice because its unreadable(?) and not easy to maintain(?)
+
+console.log(badCode);
+// £24
 
 /* BETTER PRACTICE */
 
-let goodCode =
+const pipe = (functionOne, functionTwo, functionThree) => (functionOneParameterOne, functionOneParameterTwo, functionTwoParameterTwo) => functionThree(functionTwo(functionOne(functionOneParameterOne, functionOneParameterTwo), functionTwoParameterTwo));
+let goodCode = pipe(add, multiply, format)(startingValue, 10, 2);
+// i don't neccessarily agree that pipe/piping is good code...
```

Comment:
I agree this is much less readable.

A pipe function can be good when you have multiple functions that take one argument that is the output of the last but as you can see from above its not the best when the functions take multiple arguments.

I think what we were after here is something more simple like this:
```
const sum = add(startingValue, 10);
const product = multiply(sum, 2);
const goodCode = format(product);
```

The tricky thing is thinking of good names

Reply:
When I was researching piping it was all about how you make an overall piping function that handles everything, 
and how you are then able to use the two sets of "parameter" parentheses back to back... (which I had not encountered up until that point)

That's why I did it the way I did. After learning how to do it like that^

I think that is far more difficult to wrap my head around than the example you gave of what they were looking for

So is my understanding of what piping is incorrect?😕❓ 
Is the example you gave also "piping" and whats the difference between the two? 🤔💭